Makefile

```diff
@@ -99,8 +97,8 @@ $(BINDIR)/deepcopy-gen: cmd/libs/go2idl/deepcopy-gen
 	$(DOCKER) glide install --strip-vendor
 	touch $@
 
-.scBuildImage: hack/Dockerfile
-	sed "s/GO_VERSION/$(GO_VERSION)/g" < hack/Dockerfile | \
+.scBuildImage: build/build-image/Dockerfile
+	sed "s/GO_VERSION/$(GO_VERSION)/g" < build/build-image/Dockerfile | \
```

Comment:
I understand you, but I have to disagree if only for the fact that this is intended to be an _incremental_ step toward looking more like k8s. k8s has both `build` and `hack` dirs, so I imagine we will as well. I have an imperfect understanding of the delineation between those two, but intuitively, I would guess (or hope?) that build-related stuff is isolated to `build` and that much of what is in `hack` are scripts that exist more for the sake of developer convenience? Maybe @pmorie can shed more light on this...

The bottom line (from my own point of view, at least) is that I imagine _more_ stuff moving into the `build` dir over time... we just haven't identified what those things are yet, but in recognition of it, the Dockerfile was preemptively placed where it is, which at least does move us in the direction of looking more like k8s.